### PR TITLE
fix issue on NumberFormat compact with custom numeric class

### PR DESCRIPTION
**Pull request by a14n**
Some users of my [decimal](https://pub.dev/packages/decimal) package asked to [allow using NumberFormat with Decimal objects](https://github.com/a14n/dart-decimal/issues/35). To support this I introduced an adapter DecimalIntl to be able to use a Decimal with `formatter.format(number)` (See [README](https://github.com/a14n/dart-decimal/tree/272d5835b82f688537a43c1c494f37fcf61b9e22#formatting-with-intl) for an example).

Testing it with several NumberFormat I faced an issue with `NumberFormat.compact()` because of a missing `.toInt()` that leads to an exception du to a call to `int.+(num other)` with a Decimal adapter class. This PR fix this issue.

### CODE REVIEW

File: test/custom_number_test.dart [hunk 1]
```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2014, the Dart project authors.  Please see the AUTHORS file
```
Comment (mosuem) [inline test/custom_number_test.dart:1]: Perhaps rename this file to something a bit more expressive? `custom_number_test` could be anything.

Comment (a14n) [reply test/custom_number_test.dart:1]: It looks consistent with the content of the test that uses a `MyNum` class.

If you have better name in mind, I'm open to suggestions.

Comment (mosuem) [reply test/custom_number_test.dart:1]: Maybe just `customnumber_test.dart`? I first read it as custom numbertest, which confused me. But I am just bikeshedding right now I think.

**Comment by mosuem** [comment]
Tests need to be fixed. Otherwise LGTM.

File: test/custom_number_test.dart (hunk 1 above)
Comment (mosuem) [reply test/custom_number_test.dart:1]: (You can leave it as-is, I don't know a better name either.)

**Review by mosuem — APPROVED** [review]

**Comment by a14n** [comment]
humm, please wait before merging... I think something is wrong with my change.

**Comment by a14n** [comment]
I don't find a correct solution to make compact number format working with `Decimal`. It would be great to have a kind of interface defining the methods used from this file to work with custom numbers. For now I'll will close this PR. Sorry for the noise.

**Pull request closed without merging**